### PR TITLE
New article: The Christmas Time Machine

### PR DESCRIPTION
**Pull request by thibaultduponchelle**
Hello,

This is a new article about `Feature::Compat::Try`.

Preview:
time-machine

### CODE REVIEW

File: 2022/incoming/the-christmas-time-machine.pod [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+=head1 The Christmas time machine
+
+=head2 Terrible flashback at the line work
+
+At North Pole, elves are just starting the Christmas 2022 preparations.
+
+As usual they restarted doing line work, building toys, then sorting them per area. Everyone happily singing and working! When suddenly one elf put the first toy in a box marked "Salt Lake City" and got surprised it was not empty...
+
+And he got a terrible flashbak, they seemed to have forgotten to deliver 2021 toys in Salt Lake City!
```
Comment (simbabque) [inline 2022/incoming/the-christmas-time-machine.pod:15]: - flashback, missing `c`
- to deliver _the_ 2021 toys _to_ Salt Lake City!

File: 2022/incoming/the-christmas-time-machine.pod [hunk 2]
```diff
@@ -0,0 +1,81 @@
+Author: Thibault DUPONCHELLE <thibault.duponchelle@gmail.com>
+Title: The Christmas time machine
+Topic: Feature::Compat::Try
+
+=encoding utf8
+
+=head1 The Christmas time machine
+
+=head2 Terrible flashback at the line work
```
Comment (simbabque) [inline 2022/incoming/the-christmas-time-machine.pod:9]: Can you please explain what the word flashback means? I understand this like in a movie, when you see a brief bit of what happened in the past. Is that what you mean?

File: 2022/incoming/the-christmas-time-machine.pod [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+
+At North Pole, elves are just starting the Christmas 2022 preparations.
```
Comment (simbabque) [inline 2022/incoming/the-christmas-time-machine.pod:11]: Pole, _the_ elves

File: 2022/incoming/the-christmas-time-machine.pod [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+
+As usual they restarted doing line work, building toys, then sorting them per area. Everyone happily singing and working! When suddenly one elf put the first toy in a box marked "Salt Lake City" and got surprised it was not empty...
+
+And he got a terrible flashbak, they seemed to have forgotten to deliver 2021 toys in Salt Lake City!
+
+=head2 Go ask Santa
+
+The elves were panicked, hence they reached quickly Santa for a solution.
```
Comment (simbabque) [inline 2022/incoming/the-christmas-time-machine.pod:19]: The elves _panicked_, _so_ they _quickly_ reached _out to_ Santa for a solution.

File: 2022/incoming/the-christmas-time-machine.pod [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+
+As usual they restarted doing line work, building toys, then sorting them per area. Everyone happily singing and working! When suddenly one elf put the first toy in a box marked "Salt Lake City" and got surprised it was not empty...
+
+And he got a terrible flashbak, they seemed to have forgotten to deliver 2021 toys in Salt Lake City!
+
+=head2 Go ask Santa
+
+The elves were panicked, hence they reached quickly Santa for a solution. 
+
+He took some time to think about it then had an idea: Let's use the time machine to come back in 2021 and deliver the gifts!
```
Comment (simbabque) [inline 2022/incoming/the-christmas-time-machine.pod:21]: - ... think about it_,_ then had
- to _go_ back _to_ 2021

File: 2022/incoming/the-christmas-time-machine.pod [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+
+(if you wonder: YES, time machines do exist in North Pole! And I bet that elves even already used it in the past for similar mistakes? :P)
+
+So a group of Elves decided to use the time machine, fixed the arrival date to 25 December 2021, pushed the "go" button and zooooooooooooop!
```
Comment (simbabque) [inline 2022/incoming/the-christmas-time-machine.pod:25]: Is it elves or Elves? I feel capitalising is better, but you'll have to do it for all of them.

File: 2022/incoming/the-christmas-time-machine.pod [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+	    deliver();
+	} catch($e) {
+	    print "Failed to deliver\n";
+	}
+
+=end perl
+
+But it failed with:
+
+	Feature "try" is not supported by Perl 5.30.0 at deliver.pl line 3.
```
Comment (simbabque) [inline 2022/incoming/the-christmas-time-machine.pod:50]: Slight continuity problem... 5.30 was released in 2019. Would they not have 5.34 in 2021?

**Review by simbabque — CHANGES_REQUESTED** [review]
This is very nice! I did not realise where the story was going, like it a lot. I've pointed out a couple of typos and a bit of grammar. I'm not a native speaker, so maybe someone else can look it over as well.

**Comment by thibaultduponchelle** [comment]
Thank you a lot @simbabque! I'm going to do the edits 👍🏼

File: 2022/incoming/the-christmas-time-machine.pod (hunk 7 above)
Comment (thibaultduponchelle) [reply 2022/incoming/the-christmas-time-machine.pod:50]: I used the one that is shipped with Ubuntu in 2020 

Perl 5.34 has already the `try` feature...

**Comment by oalders** [comment]
@thibaultduponchelle let me know when the edits are done and I'll look at this as well.

File: 2022/incoming/the-christmas-time-machine.pod (hunk 1 above)
Comment (thibaultduponchelle) [reply 2022/incoming/the-christmas-time-machine.pod]: Edited

File: 2022/incoming/the-christmas-time-machine.pod (hunk 4 above)
Comment (thibaultduponchelle) [reply 2022/incoming/the-christmas-time-machine.pod]: Edited

File: 2022/incoming/the-christmas-time-machine.pod (hunk 5 above)
Comment (thibaultduponchelle) [reply 2022/incoming/the-christmas-time-machine.pod]: Edited

File: 2022/incoming/the-christmas-time-machine.pod (hunk 2 above)
Comment (thibaultduponchelle) [reply 2022/incoming/the-christmas-time-machine.pod:9]: Exactly

File: 2022/incoming/the-christmas-time-machine.pod (hunk 3 above)
Comment (thibaultduponchelle) [reply 2022/incoming/the-christmas-time-machine.pod]: Edited

**Comment by thibaultduponchelle** [comment]
@oalders edits done

File: 2022/incoming/the-christmas-time-machine.pod (hunk 2 above)
Comment (openstrike) [reply 2022/incoming/the-christmas-time-machine.pod:9]: "flashback" is fine for me, instead it is "at the line work" which reads oddly. Maybe it is an Americanism?

File: 2022/incoming/the-christmas-time-machine.pod [hunk 8]
```diff
@@ -0,0 +1,81 @@
...
+
+=encoding utf8
+
+=head1 The Christmas time machine
+
+=head2 Terrible flashback at the line work
+
+At North Pole, the elves are just starting the Christmas 2022 preparations.
+
+As usual they restarted doing line work, building toys, then sorting them per area. Everyone happily singing and working! When suddenly one elf put the first toy in a box marked "Salt Lake City" and got surprised it was not empty...
```
Comment (openstrike) [inline 2022/incoming/the-christmas-time-machine.pod:13]: s/got/was/

File: 2022/incoming/the-christmas-time-machine.pod [hunk 9]
```diff
@@ -0,0 +1,81 @@
...
+
+And he got a terrible flashback, they seemed to have forgotten to deliver the 2021 toys to Salt Lake City!
```
Comment (openstrike) [inline 2022/incoming/the-christmas-time-machine.pod:15]: s/got/had/

File: 2022/incoming/the-christmas-time-machine.pod [hunk 10]
```diff
@@ -0,0 +1,81 @@
...
+
+=head2 Go ask Santa
+
+The elves panicked, so they quickly reached out Santa for a solution.
```
Comment (openstrike) [inline 2022/incoming/the-christmas-time-machine.pod:19]: s/out Santa/out to Santa/

File: 2022/incoming/the-christmas-time-machine.pod [hunk 11]
```diff
@@ -0,0 +1,81 @@
...
+
+As usual they restarted doing line work, building toys, then sorting them per area. Everyone happily singing and working! When suddenly one elf put the first toy in a box marked "Salt Lake City" and got surprised it was not empty...
+
+And he got a terrible flashback, they seemed to have forgotten to deliver the 2021 toys to Salt Lake City!
+
+=head2 Go ask Santa
+
+The elves panicked, so they quickly reached out Santa for a solution. 
+
+He took some time to think about it, then had an idea: Let's use the time machine to go back in 2021 and deliver the gifts!
```
Comment (openstrike) [inline 2022/incoming/the-christmas-time-machine.pod:21]: s/in 2021/to 2021/

File: 2022/incoming/the-christmas-time-machine.pod [hunk 12]
```diff
@@ -0,0 +1,81 @@
...
+
+(if you wonder: YES, time machines do exist in North Pole! And I bet that elves even already used it in the past for similar mistakes? :P)
+
+So a group of elves decided to use the time machine, fixed the arrival date to 25 December 2021, pushed the "go" button and zooooooooooooop!
+
+=head2 Try to deliver back in the past
+
+When arrived in 2021, they immediately tried to run Perl software that they took with them.
```
Comment (openstrike) [inline 2022/incoming/the-christmas-time-machine.pod:29]: Either "When they arrived" or "Upon arriving" would be a little better here.

File: 2022/incoming/the-christmas-time-machine.pod [hunk 13]
```diff
@@ -0,0 +1,81 @@
...
+
+For the record, in 2021, North Pole was equiped with Ubuntu 20.04 and here is the Perl code they had:
```
Comment (openstrike) [inline 2022/incoming/the-christmas-time-machine.pod:31]: "equipped" has 2 p's

**Review by openstrike — CHANGES_REQUESTED** [review]
There's nothing major wrong here AFAICS but a few minor fixes which would improve the overall reading of the article. HTH.

**Comment by thibaultduponchelle** [comment]
Thank you a lot @openstrike I will do the edits 👍🏼

File: 2022/incoming/the-christmas-time-machine.pod (hunk 8 above)
Comment (thibaultduponchelle) [reply 2022/incoming/the-christmas-time-machine.pod]: fixed

File: 2022/incoming/the-christmas-time-machine.pod (hunk 11 above)
Comment (thibaultduponchelle) [reply 2022/incoming/the-christmas-time-machine.pod]: fixed

File: 2022/incoming/the-christmas-time-machine.pod (hunk 12 above)
Comment (thibaultduponchelle) [reply 2022/incoming/the-christmas-time-machine.pod]: fixed

File: 2022/incoming/the-christmas-time-machine.pod (hunk 13 above)
Comment (thibaultduponchelle) [reply 2022/incoming/the-christmas-time-machine.pod]: fixed

**Comment by thibaultduponchelle** [comment]
Thank you a lot @openstrike for the corrections 👍🏼 And thank you again @simbabque for first round

@oalders I did the edits, ready for your review now 😃

File: 2022/incoming/the-christmas-time-machine.pod [hunk 14]
```diff
@@ -0,0 +1,81 @@
+Author: Thibault DUPONCHELLE <thibault.duponchelle@gmail.com>
+Title: The Christmas time machine
```
[the remark this answers is not in the document]
Comment (oalders) [reply 2022/incoming/the-christmas-time-machine.pod:2]: @thibaultduponchelle this is still outstanding ^^

File: 2022/incoming/the-christmas-time-machine.pod [hunk 15]
```diff
@@ -0,0 +1,81 @@
...
+Topic: Feature::Compat::Try
+
+=encoding utf8
+
+=head1 The Christmas time machine
```
Comment (oalders) [inline 2022/incoming/the-christmas-time-machine.pod:7]:
```suggestion
=head1 The Christmas Time Machine
```

File: 2022/incoming/the-christmas-time-machine.pod [hunk 16]
```diff
@@ -0,0 +1,81 @@
...
+
+=head2 A terrible flashback
+
+At North Pole, the elves are just starting the Christmas 2022 preparations.
```
Comment (oalders) [inline 2022/incoming/the-christmas-time-machine.pod:11]:
```suggestion
At the North Pole, the elves are just starting the Christmas 2022 preparations.
```

File: 2022/incoming/the-christmas-time-machine.pod [hunk 17]
```diff
@@ -0,0 +1,81 @@
...
+
+And he had a terrible flashback, they seemed to have forgotten to deliver the 2021 toys to Salt Lake City!
+
+=head2 Go ask Santa
+
+The elves panicked, so they quickly reached out to Santa for a solution. 
+
+He took some time to think about it, then had an idea: Let's use the time machine to go back to 2021 and deliver the gifts!
+
+(if you wonder: YES, time machines do exist in North Pole! And I bet that elves even already used it in the past for similar mistakes? :P)
```
Comment (oalders) [inline 2022/incoming/the-christmas-time-machine.pod:23]:
```suggestion
(If you're wondering: YES, time machines do exist in North Pole! And I bet that elves even already used it in the past for similar mistakes? :P)
```

**Review by oalders — CHANGES_REQUESTED** [review]
Love it! Just a few small suggestions.

**Comment by thibaultduponchelle** [comment]
Fixed all remarks, squashed.

Ready to merge! @oalders 

See the new preview:
the-christmas-time-machine-preview

**Comment by thibaultduponchelle** [comment]
Again, thank you a lot @simbabque and @openstrike 🙏🏼

**Comment by oalders** [comment]
I'm going to close and re-open to see if that runs the test workflow.

**Review by oalders — APPROVED** [review]

**Comment by oalders** [comment]
@openstrike and @simbabque, you'll need to approve the changes in order for us to merge this PR.

**Comment by thibaultduponchelle** [comment]
@openstrike @simbabque can you please approve? 🙏🏼

File: 2022/incoming/the-christmas-time-machine.pod [hunk 18]
```diff
@@ -0,0 +1,81 @@
+Author: Thibault DUPONCHELLE <thibault.duponchelle@gmail.com>
+Title: The Christmas Time Machine
+Topic: Feature::Compat::Try
+
+=encoding utf8
+
+=head1 The Christmas Time Machine
```
Comment (oalders) [inline 2022/incoming/the-christmas-time-machine.pod:7]:
```suggestion
```

File: 2022/incoming/the-christmas-time-machine.pod [hunk 19]
```diff
@@ -0,0 +1,81 @@
...
+
+=head2 A terrible flashback
```
Comment (oalders) [inline 2022/incoming/the-christmas-time-machine.pod:9]:
```suggestion
=head2 A Terrible Flashback
```

File: 2022/incoming/the-christmas-time-machine.pod [hunk 20]
```diff
@@ -0,0 +1,81 @@
...
+
+At the North Pole, the elves are just starting the Christmas 2022 preparations.
+
+As usual they restarted doing line work, building toys, then sorting them per area. Everyone happily singing and working! When suddenly one elf put the first toy in a box marked "Salt Lake City" and was surprised it was not empty...
+
+And he had a terrible flashback, they seemed to have forgotten to deliver the 2021 toys to Salt Lake City!
+
+=head2 Go ask Santa
```
Comment (oalders) [inline 2022/incoming/the-christmas-time-machine.pod:17]:
```suggestion
=head2 Go Ask Santa
```

File: 2022/incoming/the-christmas-time-machine.pod [hunk 21]
```diff
@@ -0,0 +1,81 @@
...
+
+The elves panicked, so they quickly reached out to Santa for a solution. 
+
+He took some time to think about it, then had an idea: Let's use the time machine to go back to 2021 and deliver the gifts!
+
+(If you're wondering: YES, time machines do exist in North Pole! And I bet that elves even already used it in the past for similar mistakes? :P)
+
+So a group of elves decided to use the time machine, fixed the arrival date to 25 December 2021, pushed the "go" button and zooooooooooooop!
+
+=head2 Try to deliver back in the past
```
Comment (oalders) [inline 2022/incoming/the-christmas-time-machine.pod:27]:
```suggestion
=head2 Try to Deliver Back in the Past
```

**Review by oalders — APPROVED** [review]

**Pull request merged**